python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py

```diff
@@ -244,29 +243,27 @@ def on_message(client, userdata, msg):
         if _callback_event_loop is None:
             asyncio.set_event_loop(asyncio.new_event_loop())
             _callback_event_loop = asyncio.get_event_loop()
-        time_stamp = utils.local_timestamp()
         # Protobuf message structure
-        readings_list = sparkplug_b_pb2.ReadingsList()
-        readings_list.ParseFromString(msg.payload)
-        for reading in readings_list.readings:
-            # Initialize the structured data dictionary
+        sparkplug_payload = sparkplug_b_pb2.Payload()
+        sparkplug_payload.ParseFromString(msg.payload)
+
+        for metric in sparkplug_payload.metrics:
+            value = ""
+            if metric.HasField("bool_value"):
+                value = metric.bool_value
```

Comment:
cast to int?

Reply:
The C layer has consistently stored values as 1 and 0. More info available at FOGL-8067